src/components/batch-models-multi-select/CopyInUserBatchModelsMultiSelect.component.js

```diff
@@ -0,0 +1,224 @@
+                    state: "ready",
+                    parents: parentsLoaded,
+                    allChildren,
+                    selectedIds: this.props.model
+                        .getSelectedChildren(parentsLoaded)
+                        .map(obj => obj.id),
+                })
+            )
+            .catch(err =>
+                this.close(this.getTranslation("error_loading_data") + " :" + err.toString())
+            );
+    }
+
+    close(snackMessage = null) {
+        if (snackMessage) snackActions.show({ message: snackMessage });
+        this.props.onRequestClose();
+    }
+
+    renderStrategyToggle() {
+        if (this.state.parents && this.state.parents.length > 1) {
+            const label =
+                this.getTranslation("update_strategy") +
+                ": " +
+                this.getTranslation("update_strategy_" + this.state.updateStrategy);
+
+            return (
+                <Toggle
+                    label={label}
+                    style={this.styles.toggle}
+                    checked={this.state.updateStrategy === "replace"}
+                    onToggle={(ev, newValue) =>
+                        this.setState({ updateStrategy: newValue ? "replace" : "merge" })
+                    }
+                />
+            );
+        } else {
+            return null;
+        }
+    }
+    async copyInUserSave() {
+        const { parents, selectedIds, copyUserGroups, copyUserRoles } = this.state;
+        await this.props.model
+            .copyInUserSave(parents, selectedIds, copyUserGroups, copyUserRoles)
+            .then(() => this.close(this.props.onSuccess))
+            .catch(err => this.close(this.props.onError));
+    }
+    onChange(selectedIds) {
+        this.setState({ selectedIds });
+    }
+    render() {
+        switch (this.state.state) {
+            case "loading":
+                return (
+                    <div style={this.styles.loadingMask}>
+                        <LoadingMask />
+                    </div>
+                );
+            case "error":
+                return <div style={this.styles.loadingMask}>{this.state.error}</div>;
+            case "ready":
+                return this.renderForm();
+            default:
+                throw new Error(`Unknown state: ${state}`);
+        }
+    }
+
+    onFilterTextChange(event) {
+        this.setState({ filterText: event.target.value });
+    }
+    getDialogButtons() {
+        return [
+            <FlatButton
+                label={this.getTranslation("cancel")}
+                onClick={this.props.onRequestClose}
+                style={this.styles.cancelButton}
+            />,
+            <RaisedButton
+                primary
+                label={this.getTranslation("save")}
+                onClick={ this.state.copyUserGroups || this.state.copyUserRoles ? this.copyInUserSave.bind(this) : err => this.close("Error: You must select a toggle option") }
+            />,
```

Comment:
I was originally trying to have an error message show in the modal where if there is an error, it changes the state of an errorMessage prop but I was having issues with that because it didn't like that I was calling a function in an eventListener that was changing state. I will try to integrate a snackbar instead within the modal.

Reply:
It's ok for the snackbar to show at the bottom. Let's just keep the modal open so the user can go on with the operation.